runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java

```diff
@@ -72,7 +81,26 @@ public DefaultIllumina(Builder builder) {
     return Arrays.stream(root.listFiles(f -> f.isDirectory() && !f.getName().equals("Instrument")));
   }
 
-  private static final Pattern COMMA = Pattern.compile(",");
+  private boolean isLaneComplete(Path laneDir, int numCycles) {
+    try {
+      // First, examine the control files to determine all the BCL files we intend to find for each cycle.
+      List<String> targetBclFileNames = Files.walk(laneDir, 1)//
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "Stream" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)